src/Items/ItemDye.h

```diff
@@ -118,6 +118,11 @@ class cItemDyeHandler :
 			{
 				// Grow by 2 - 5 stages:
 				auto numStages = GetRandomProvider().RandInt(2, 5);
+				// No longer create a melon / pumpkin block when right-clicking on a grown melon / pumpkin seed
```

Comment:
We should not do nothing here, we only need to cap the numStages in the call to GrowPlantAt to be min(7-currentGrowth, numStages)

Reply:
This is to prevent creating a melon / pumpkin block when someone right-click on a full grown plant with bone meal.
This prevents also dispenser to do it.